common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -647,3 +653,28 @@ def associate_by_email_if_login_api(auth_entry, backend, details, user, *args, *
             # email address and the legitimate user would now login to the illegitimate
             # account.
             return association_response
+
+
+@partial.partial
+def create_sudo_session(strategy, auth_entry, *args, **kwargs):
```

Comment:
I'm not wild about `third_party_auth` having to explicitly know about sudo mode. Could this method be added as part of a micro package that handles the intersection between these features, and injects the function into the `SOCIAL_AUTH_PIPELINE` on startup?